### DISCUSSION TITLE
Problem with Suspension Simulation: Cable Not Showing Stretch Under Load

### DISCUSSION THREAD

**Discussion by ValentinaGiannotti** [google-deepmind/mujoco#2273] [Asking for Help]
### Intro

Hi!

I am a  student  at University of Federico II, I use MuJoCo for my research .

### My setup

Mujoco 3.2.6

### My question

Hi everyone,

I'm working on implementing a suspension system in MuJoCo to analyze the oscillations of an object hanging from a cable. I've modeled the cable as a composite object of type cable and added a constraint to attach the object to the cable. However, I'm encountering an issue: while the object remains suspended, suggesting that the cable stretches due to the weight, this deformation is not visually represented in the simulation.

I’ve tried tuning the equality solref parameters, but the object continues to float as if disconnected. Additionally, the heavier the object, the further it drifts away from the cable.

### Minimal model and/or code that explain my question

<mujoco model="Cable">
<compiler angle="radian" eulerseq="XYZ" coordinate="local"/>
  <include file="scene.xml"/>

  <extension>
    <plugin plugin="mujoco.elasticity.cable"/>
  </extension>

  <statistic center="0 0 0" extent="1"/>
  <visual>
    <global elevation="-30"/>
  </visual>

  <compiler autolimits="true"/>
  <option gravity="0 0 -9.81"/>

  <size memory="2M"/>

  <worldbody>
    <body name="Cable" pos=" 0 0 2">
        <composite type="cable" curve="s" count="20 1 1" size="1" offset="0 0 0"  initial="ball">
        <plugin plugin="mujoco.elasticity.cable">
            
            <config key="twist" value="1000"/>
            <config key="bend" value="500"/>
            <config key="vmax" value="20000"/>
            <config key="flat" value="true"/>
        </plugin>
        <joint kind="main" damping=".015"/>
        <joint kind="stretch" stiffness="1000" damping="5" range="0.9 1.5"/>

        <geom type="capsule" size=".005" rgba=".8 .2 .1 1" condim="1"/>
        <site group="0"/>
        </composite>
         <inertial pos="0 0 0" mass="8" quat="1 0 0 0" diaginertia="0.01167 0.01167 0.01"/>
        <joint name="crane_x" type="slide" damping="100" axis="1 0 0"/>
        <joint name="crane_y" type="slide" damping="100" axis="0 1 0"/>
    </body>
    <body name="slider" pos="0 0 1" euler="0 0 -1.57">
        <geom type="cylinder" size="0.05 0.05"/>
        <site name="s2" pos="0 0 0.05"/>
        <inertial pos="0 0 0" mass="15" quat="1 0 0 0" diaginertia="0.01167 0.01167 0.01"/>
        <joint type="free" damping="0.01"/>
    </body>
  </worldbody>

  <equality>
    <weld site1="S_last" site2="s2" solimp="1 0.9 1.75 0.2"/>
  </equality>

  <actuator>
    <position name="crane_x" joint="crane_x" ctrlrange="-1 1" kp="100"/>
    <position name="crane_y" joint="crane_y" ctrlrange="-1 1" kp="100"/>
  </actuator>
</mujoco>

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by quagla** [comment]
Hi @ValentinaGiannotti the cable cannot stretch, it's meant for situation where the tensile load is negligible compared to the other forces. If you have a cable in a tensile configuration, I would recommend using flexcomp instead. I have added an extensible 1D example last week https://github.com/google-deepmind/mujoco/blob/main/model/flex/pulley.xml (note that I haven't had the time to add bending and twist stiffness to this extensible cable yet)

**Reply by ValentinaGiannotti** [reply]
Thank you so much for the suggestion and for sharing the extensible 1D example! This solution has been helpful in approximating the desired behavior, but it’s not yet entirely perfect for the specific issue I’m working on. I was wondering: would it be possible to use this feature to create a rope-like behavior instead of a closed chain (circle)? I believe this approach might better suit the requirements of the project. Thanks in advance for your time and support!

**Reply by quagla** [reply]
I'm not completely sure about the question but I believe what you want to do is to replace `type="circle"` with `type="grid"`?

**Reply by ValentinaGiannotti** [reply]
Yes, exactly. Using the grid approach, the issue is that I cannot attach the first and last bodies of the cable to the cylinder. I tried using the weld command to link the bodies, but it doesn't seem to work as expected. Additionally, the FlexComp command does not allow me to create a site.
What I want to achieve is for the cable to wrap around the first cylinder while still being able to support a weight.
[cable.zip](https://github.com/user-attachments/files/18082755/cable.zip)

**Reply by quagla** [reply]
What is the issue with the weld?

**Reply by ValentinaGiannotti** [reply]
When I use the weld to connect two bodies, I can make them move together, but not as a single well-defined entity. This happens because I can’t determine exactly where the connection between the two bodies takes place. I would like to use sites to precisely define the connection point, but this is not possible because I can’t define a site for flexcomp. As a result, I can’t accurately control the specific point where the cable and the cylinder are connected.

**Reply by quagla** [reply]
You can load and save the XML and the flexcomp will be expanded into bodies, to which you can add a site. Is this a feasible solution?

**Reply by ValentinaGiannotti** [reply]
Thanks, it works
